### ISSUE TITLE
FN: `use_self` doesn't check `pat`

### ISSUE THREAD

**Issue by Y-Nak** [rust-lang/rust-clippy#6955]
Lint name: `use_self`

I tried this code:

```rust
#![deny(clippy::use_self)]
#![allow(unused)]

enum Foo {
    Variant,
}

impl Foo {
    pub fn foo(self) {
        match self {
            Foo::Variant => {}
        }
    }
}
```

I expected to see this happen: the usage of `Foo::Variant` triggers ` use_self`.

Instead, this happened: Nothing.

### Meta

- `cargo clippy -V`: clippy 0.1.52 (5d04957 2021-03-22)

- `rustc -Vv`:
```
rustc 1.53.0-nightly (5d04957a4 2021-03-22)
binary: rustc
commit-hash: 5d04957a4b4714f71d38326fc96a0b0ef6dc5800
commit-date: 2021-03-22
host: x86_64-unknown-linux-gnu
release: 1.53.0-nightly
LLVM version: 12.0.0
```
